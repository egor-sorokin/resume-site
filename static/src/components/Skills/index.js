import React, { Component } from 'react';
import withFetching from '../../utils/api';
import { URL_PATH_SKILLS } from '../../constants/index';


const Skills = ({ data, isFetching, error }) => {
  const skills = data.skills || [];

  if (error) {
    return <p>{error.message}</p>;
  }

  if (isFetching) {
    return <p>Loading ...</p>;
  }

  return (
    <div>
      dasd
      <div>
        {skills.map(item =>
          <div key={item.id}>
            asd
          </div>
        )}
      </div>
    </div>
  );
};

export default withFetching(URL_PATH_SKILLS, Skills);