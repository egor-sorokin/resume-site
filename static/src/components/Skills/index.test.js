import React from 'react';
import Skills from './index';
import { shallow, render, mount } from 'enzyme';

describe('<Skills />', () => {
  it('renders 1 <Skills /> component', () => {
    const component = shallow(<Skills />);
    expect(component).toHaveLength(1);
  })
});