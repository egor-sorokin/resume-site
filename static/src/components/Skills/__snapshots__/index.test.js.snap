// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> matches snapshot 1`] = `
<Skills
  data={Object {}}
  error={null}
  isFetching={true}
/>
`;
